ignite/engine/state.py

```diff
@@ -64,18 +67,26 @@ def __init__(self, **kwargs: Any) -> None:
         for k, v in kwargs.items():
             setattr(self, k, v)
 
-        self._update_attrs()
+        # Update the attributes if it's a standalone state
+        # to keep BC compatibility
+        if self.engine is None:
+            self._update_attrs()
 
     def _update_attrs(self) -> None:
-        for value in self.event_to_attr.values():
-            if not hasattr(self, value):
-                setattr(self, value, 0)
+        for key in self.attr_to_events.keys():
+            if not hasattr(self, key):
+                setattr(self, key, 0)
 
     def get_event_attrib_value(self, event_name: Union[str, Events, CallableEventWithFilter]) -> int:
```

Comment:
Let's deprecate this method as now it is engine's responsibility

Reply:
I'm not sure about doing this now, I suggest leave this part for now as didn't think of a replacement yet, check this https://github.com/pytorch/ignite/search?q=get_event_attrib_value, it's used in engine/handlers/loggers/tests/..